core/kazoo_documents/src/kzd_ledgers.erl

```diff
@@ -255,19 +259,19 @@ set_account_name(Doc, AccountName) ->
 %% @doc
 %% @end
 %%------------------------------------------------------------------------------
--spec unit_amount(doc()) -> kz_term:api_integer().
+-spec unit_amount(doc()) -> kz_term:api_number().
 unit_amount(Doc) ->
     unit_amount(Doc, 'undefined').
 
 -spec unit_amount(doc(), Default) -> kz_currency:units() | Default.
 unit_amount(Doc, Default) ->
-    kz_json:get_integer_value(?AMOUNT, Doc, Default).
+    kz_json:get_number_value(?AMOUNT, Doc, Default).
 
 %%------------------------------------------------------------------------------
 %% @doc
 %% @end
 %%------------------------------------------------------------------------------
--spec set_unit_amount(doc(), integer()) -> doc().
+-spec set_unit_amount(doc(), number()) -> doc().
```

Comment:
@jamesaimonetti actually, the second argument should be kz_currency:units().  Same as above tho, would not expect anything other than an integer here.